tests/server-common/src/test/java/io/a2a/server/apps/common/AbstractA2AServerTest.java

```diff
@@ -370,517 +241,41 @@ public void testError() {
-                    .build();
-            SendStreamingMessageRequest request = new SendStreamingMessageRequest(
-                    "1", new MessageSendParams(message, null, null));
-
-            CompletableFuture<HttpResponse<Stream<String>>> responseFuture = initialiseStreamingRequest(request, null);
-
-            CountDownLatch latch = new CountDownLatch(1);
-            AtomicReference<Throwable> errorRef = new AtomicReference<>();
-
-            responseFuture.thenAccept(response -> {
-                if (response.statusCode() != 200) {
-                    //errorRef.set(new IllegalStateException("Status code was " + response.statusCode()));
-                    throw new IllegalStateException("Status code was " + response.statusCode());
-                }
-                response.body().forEach(line -> {
-                    try {
-                        SendStreamingMessageResponse jsonResponse = extractJsonResponseFromSseLine(line);
-                        if (jsonResponse != null) {
-                            assertNull(jsonResponse.getError());
-                            Message messageResponse = (Message) jsonResponse.getResult();
-                            assertEquals(MESSAGE.getMessageId(), messageResponse.getMessageId());
-                            assertEquals(MESSAGE.getRole(), messageResponse.getRole());
-                            Part<?> part = messageResponse.getParts().get(0);
-                            assertEquals(Part.Kind.TEXT, part.getKind());
-                            assertEquals("test message", ((TextPart) part).getText());
-                            latch.countDown();
-                        }
-                    } catch (JsonProcessingException e) {
-                        throw new RuntimeException(e);
-                    }
-                });
-            }).exceptionally(t -> {
-                if (!isStreamClosedError(t)) {
-                    errorRef.set(t);
-                }
-                latch.countDown();
-                return null;
-            });
-
-            boolean dataRead = latch.await(20, TimeUnit.SECONDS);
-            Assertions.assertTrue(dataRead);
-            Assertions.assertNull(errorRef.get());
-        } catch (Exception e) {
-        } finally {
-            getTaskStore().delete(MINIMAL_TASK.getId());
+            SendMessageResponse response = client.sendMessage(messageSendParams);
+            assertNull(response.getResult());
+            // this should be an instance of UnsupportedOperationError, see https://github.com/a2aproject/a2a-java/issues/23
+            assertInstanceOf(JSONRPCError.class, response.getError());
+            assertEquals(new UnsupportedOperationError().getCode(), response.getError().getCode());
+        } catch (A2AServerException e) {
+            // Handle exception
         }
     }
 
     @Test
-    public void testResubscribeExistingTaskSuccess() throws Exception {
```

Comment:
We shouldn't delete this. We should be able to tweak it to use an A2A client method. Let us know if you have questions about this.